src/regex_impl.cc

```diff
@@ -328,6 +328,14 @@ struct RegexParser
                 if (contains(StringView{"^$.*+?[]{}"}, cp) or (cp >= 0xF0000 and cp <= 0xFFFFF))
                     parse_error(format("unexpected '{}'", cp));
                 ++m_pos;
+                if (is_upper(cp) && (m_flags & Flags::SmartCase)) {
+                  m_flags &= ~Flags::IgnoreCase;
```

Comment:
Why do we clear IgnoreCase here ? Doesn't that mean we have a different behaviour for the first upper case character ?

Reply:
Shouldn't do, the IgnoreCase isn't checked until the add_node call just after this check.